src/services/job-processor/message-storage.ts

```diff
@@ -0,0 +1,29 @@
+import { ExitMessage } from './service.js'
+
+export class MessageStorage {
+  protected $messages: ExitMessage[] = []
+  protected $signatures: Set<string> = new Set()
+
+  public addMessage(message: ExitMessage): number {
+    if (!this.$signatures.has(message.signature)) {
```

Comment:
I guess the key should be the validator index. In this case, we can replace invalid signatures with new correct ones without having to stop the ejector